tools/autograd/gen_variable_type.py

```diff
@@ -679,6 +705,9 @@ def gen_variable_type_shard(out, aten_declarations, template_path, suffix, heade
                 wrapper_registrations.append(UNBOXEDONLY_WRAPPER_REGISTRATION.substitute(
                     declaration, class_type='VariableType'))
 
+        # See Note [Manual catchAll kernels]
+        assert (declaration['name'] in MANUAL_CATCHALL) == declaration['manual_kernel_registration']
```

Comment:
Why do you need the extra `MANUAL_CATCHALL` set if it has to exactly match the ones with `declaration['manual_kernel_registration'] == True` ?

Reply:
sorry for late reply! For the ops in MANUAL_CATCHALL we used to skip both aten & variable registration (so that catchall can be used). 
For more context see https://github.com/pytorch/pytorch/pull/42386#issuecomment-668274333. Note that (4) is possible but we currently don't have a use case for it yet so I added the assert so that we don't accidentally change the behavior. In the future if there's a use case we could remove the assert as needed.